lib/gitlab/markdown.rb

```diff
@@ -64,7 +65,12 @@ def gfm_with_options(text, options = {}, project = @project, html_options = {})
         current_user:    current_user,
         only_path:       options[:reference_only_path],
         project:         project,
-        reference_class: html_options[:class]
+        reference_class: html_options[:class],
+
+        # RelativeLinkFilter
+        ref: @ref,
+        requested_path: @path,
+        project_wiki: @project_wiki
```

Comment:
If they were available to the old helper, they should be available here. If all three are truly required by the filter, you can check at the beginning of `call` and `return doc` if any of them are nil.